### PR TITLE
feat(consumer): Add support for VALUES format when writing to ClickHouse

### PR DESCRIPTION
**Pull request by onewland**
This is necessary for an upcoming change which will write aggregate expressions directly in the insert.

Functional changes:
- support for inserting using the VALUES format (https://clickhouse.com/docs/en/interfaces/formats/#data-format-values) at the WritableTableStorage level
- support in the multistorage consumer for serializing to the VALUES format (including a new ValuesRowEncoder)
- added a new type of return from processing called `AggregateInsertBatch` to indicate to callers that the data being returned needs to be interpreted by Clickhouse (e.g. it can contain function call expressions)

This was split out of https://github.com/getsentry/snuba/pull/2346 to be more readable (without the additional metrics changes). There will be an upcoming PR based off this one to support direct metrics aggregate writing

### CODE REVIEW

**Review by fpacifici — APPROVED** [review]
Seems good to me.

File: snuba/clickhouse/http.py [hunk 1]
```diff
@@ -39,12 +41,31 @@ def __default(self, value: Any) -> Any:
...
             raise TypeError
 
-    def encode(self, value: WriterTableRow) -> JSONRow:
+    def encode(self, value: WriterTableRow) -> bytes:
         return cast(
             bytes, rapidjson.dumps(value, default=self.__default).encode("utf-8")
         )
 
 
+class ValuesRowEncoder(Encoder[bytes, WriterTableRow]):
```
Comment (fpacifici) [inline snuba/clickhouse/http.py:50]: PLease add some unit tests for this before merging.

Comment (onewland) [reply snuba/clickhouse/http.py:50]: I've added a few unit tests. Let me know if you'd like coverage of anything else.

Comment (fpacifici) [reply snuba/clickhouse/http.py:50]: It is ok to me.
Optional:
I would add some more complex expression(like function containing a function). 
Also what happens if you define an alias in an expression in the INSERT? I never tried.

**Pull request merged**